99-GitConfiguration/README.md

```diff
@@ -1,154 +1,158 @@
-
-	$ git config --global core.editor 'VIM="C:/Dev/vim" HOME="C:/Users/tv185035" cyg-wrapper.sh "C:/Dev/vim/vim74/gvim.exe"'
-
-Now your GVim should use same settings when started from windows or from git.
+# Configuration
+
+## Introduction
+
+* _GIT_ stores the configuration in ```gitconfig``` files.
+* The configuration file defines the user's name and email which
+  is being used during ```git-commit```.
+* There are three levels of configuration:
+    * system
+    * Machine specific, shared with all users.
+    * To edit the file in plain text you can use:
+
+```
+git config --system --edit
+```
+
+  * global
+    * User specific, shared by given user for all repositories.
+    * Overrides _system_ level.
+    * To edit the file in plain text you can use:
+
+```
+git config --global --edit
+```
+  * local
+    * Repository specific.
+    * Overrides _global_ and _system_ level.
+    * To edit the file in plain text you can use:
+
+```
+git config --local --edit
+```
+
+## E-mail and user name configuration
+
+* Two commands below will set user's email and user name on global level.
+* This means that unless the cloned repository has a local override
+  these values will be used.
+* It is very handy if you work on multiple repositories.
+* You can always use ```--local``` instead of ```--global``` to override
+  values per repository.
+* File should be located as ```.gitconfig``` in your home directory.
+
+```
+git config --global user.email "your_email@example.com"
+```
+
+```
+git config --global user.name "Vladimir Kloz"
+```
+
+## Merge tool
+
+### Visual Studio
+
+* It is pretty handy to use a visual merge tool you are used to using.
+* For 3-way merge Visual Studio works pretty well.
+* Open the global configuration file:
+
+```
+git config --global --edit
+```
+
+* Copy & Paste lines below to it (obviously adjust paths if/as needed
+  to match your VS version/location):
+
+```
+[merge]
+      tool = vsdiffmerge
+[mergetool]
+      prompt = false
+      keepbackup = false
+[mergetool "vsdiffmerge"]
+      cmd = '"C:/Program Files (x86)/Microsoft Visual Studio 12.0/Common7/IDE/vsdiffmerge.exe"' "$REMOTE" "$LOCAL" "$BASE" "$MERGED" //m
+      trustexitcode = true
+```
+
+* Save the file.
+
+### KDiff3
+
+Alternatively, if you want to use something smaller and (imho) better, you can go with KDiff3. Opensource, multiplatform and free.
+
+#### Git for Windows
+
+	$ git config --global mergetool.kdiff3.path "C:/Program Files/KDiff3/kdiff3.exe"
+	$ git config --global merge.tool=kdiff3
+
+#### Cygwin
+
+This is little harder. It assumes you have `cyg-wrapper.sh` script in `~/bin`.
+
+	$ git config --global mergetool.kdiff3.path
+	$ git config --global mergetool.kdiff3.cmd '~/bin/cyg-wrapper.sh "C:/Program Files/KDiff3/kdiff3.exe" "$LOCAL" "$REMOTE" --output "$MERGED" --base "$BASE"'
+	$ git config --global merge.tool kdiff3
+
+#### DiffMerge

```

Comment:
I have been using "DiffMerge" for a while, so I added a line about it.  It is free and pretty basic.
Not sure how many developers use it.  There is a list of merge tools on the Confluence page here:  "http://confluence.ncr.com/display/POSDev/Setting+up+a+Development+Machine+for+POS" See section "Install Optional Software" point #5.